gradient_utils/metrics/metrics.py

```diff
@@ -6,19 +6,43 @@
 logger = logging.getLogger(__name__)
 
 
+def _get_default_grouping_key(workload_id=None, step=None):
+    return {
+        get_workload_label(): workload_id or get_workload_id(),
+        'step': step,
+    }
+
+
 def add_metrics(
-        metrics,
+        metrics_map,
         step=None,
         timeout=30):
-    metrics_logger = MetricsLogger(step=step)
-
-    metrics = [Metric(key, value) for key, value in metrics.items()]
+    metrics = [Metric(key, value) for key, value in metrics_map.items()]
+    default_grouping_key = _get_default_grouping_key(step=step)
     for metric in metrics:
-        metrics_logger.add_gauge(metric.key)
-        logger.debug("Setting metric key: %s, value: %s", metrics_logger[metric.key], metric.value)
-        metrics_logger[metric.key].set(metric.value)
-
-    metrics_logger.push_metrics(timeout)
+        registry = CollectorRegistry(auto_describe=True)
+        new_metric_raw = Gauge(
+            metric.key,
+            documentation="",
+            registry=registry,
+            labelnames=[
+                get_workload_label(),
+                "pod",
+                "step"])
+        new_metric = new_metric_raw.labels(get_workload_id(), HOSTNAME, step)
+        new_metric.set(metric.value)
+        new_grouping_key = {
+            "name": metric.key,
+        }
+        new_grouping_key.update(default_grouping_key)
```

Comment:
i think you need to pass step into default_grouping_key

Maybe default_grouping_key should be first and then name and step should be updated. this way feels more natural

Reply:
it's passed into the call to `_get_default_grouping_key` on line 21.